packages/kbn-esql-validation-autocomplete/src/shared/esql_types.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ESQLNumericLiteralType } from '@kbn/esql-ast/src/types';
+
+export const ESQL_COMMON_NUMERIC_TYPES = ['double', 'long', 'integer'] as const;
+export const ESQL_NUMERIC_DECIMAL_TYPES = [
+  'double',
+  'unsigned_long',
+  'long',
+  'counter_long',
+  'counter_double',
+] as const;
+export const ESQL_NUMBER_TYPES = [
+  'integer',
+  'counter_integer',
+  ...ESQL_NUMERIC_DECIMAL_TYPES,
+] as const;
+
+export const ESQL_STRING_TYPES = ['keyword', 'text'] as const;
+export const ESQL_DATE_TYPES = ['datetime', 'date_period'] as const;
+
+/**
+ *
+ * @param type
+ * @returns
+ */
+export function isStringType(type: unknown) {
+  return typeof type === 'string' && ['keyword', 'text'].includes(type);
+}
+
+export function isNumericType(type: unknown): type is ESQLNumericLiteralType {
+  return (
+    typeof type === 'string' &&
+    [...ESQL_NUMBER_TYPES, 'decimal'].includes(type as (typeof ESQL_NUMBER_TYPES)[number])
+  );
+}
+
+export function isNumericDecimalType(type: unknown): type is ESQLNumericLiteralType {
```

Comment:
```suggestion
export function isNumericDecimalType(type: unknown): type is ESQLDecimalLiteralType {
```

Reply:
Thanks for catching that! Updated here [`4d1b7a7` (#188600)](https://github.com/elastic/kibana/pull/188600/commits/4d1b7a71b5092a05381063d1609ec6187dedc2a8)